probe/process/reporter.go

```diff
@@ -45,12 +45,19 @@ func (r *Reporter) processTopology() (report.Topology, error) {
 	err := r.walker.Walk(func(p Process) {
 		pidstr := strconv.Itoa(p.PID)
 		nodeID := report.MakeProcessNodeID(r.scope, pidstr)
-		t.NodeMetadatas[nodeID] = report.MakeNodeMetadataWith(map[string]string{
+		metadata := map[string]string{
 			PID:     pidstr,
 			Comm:    p.Comm,
 			Cmdline: p.Cmdline,
 			Threads: strconv.Itoa(p.Threads),
-		})
+		}
+		// Filter out empty fields
+		for key, value := range metadata {
+			if value == "" {
+				delete(metadata, key)
+			}
+		}
+		t.NodeMetadatas[nodeID] = report.MakeNodeMetadataWith(metadata)
```

Comment:
This would be better if it were more similar to detailed_node rendering:
```
	for _, tuple := range []struct{ key, value string }{
		{Comm, p.Comm},
		{PID, pidstr},
		{Cmdline, p.Cmdline},
		{Threads, strconv.Itoa(p.Threads)},
	} {
		if tuple.value != "" {
			rt.NodeMetadatas[nodeID][key] = value
		}
	}
```
Or similar...